mpulse-real-user-measurement.php

```diff
@@ -86,56 +86,178 @@ function mpulse_plugin_page() {
 function mpulse_add_rum_header() {
 ?>
 <script>
-(function(){
-  // Boomerang Loader Snippet version 10
-  if (window.BOOMR && (window.BOOMR.version || window.BOOMR.snippetExecuted)) {
-    return;
-  }
-
-  window.BOOMR = window.BOOMR || {};
-  window.BOOMR.snippetExecuted = true;
-
-  var dom, doc, where, iframe = document.createElement("iframe"), win = window;
-
-  function boomerangSaveLoadTime(e) {
-    win.BOOMR_onload = (e && e.timeStamp) || new Date().getTime();
-  }
-
-  if (win.addEventListener) {
-    win.addEventListener("load", boomerangSaveLoadTime, false);
-  } else if (win.attachEvent) {
-    win.attachEvent("onload", boomerangSaveLoadTime);
-  }
-
-  iframe.src = "javascript:void(0)";
-  iframe.title = "";
-  iframe.role = "presentation";
-  (iframe.frameElement || iframe).style.cssText = "width:0;height:0;border:0;display:none;";
-  where = document.getElementsByTagName("script")[0];
-  where.parentNode.insertBefore(iframe, where);
-
-  try {
-    doc = iframe.contentWindow.document;
-  } catch (e) {
-    dom = document.domain;
-    iframe.src = "javascript:var d=document.open();d.domain='" + dom + "';void(0);";
-    doc = iframe.contentWindow.document;
-  }
-
-  doc.open()._l = function() {
-    var js = this.createElement("script");
-    if (dom) {
-      this.domain = dom;
+(function() {
+    // mPulse Loader Snippet version 12
+    if (window.BOOMR && (window.BOOMR.version || window.BOOMR.snippetExecuted)) {
+        return;
+    }
+
+    window.BOOMR = window.BOOMR || {};
+    window.BOOMR.snippetExecuted = true;
+    window.BOOMR.snippetVersion = 12;
+
+    var boomerangUrl = "//c.go-mpulse.net/boomerang/<?php echo get_option('mpulse_api_key'); ?>",
+        // document.currentScript is supported in all browsers other than IE
+        where = document.currentScript || document.getElementsByTagName("script")[0],
+        // Whether or not Preload method has worked
+        promoted = false,
+        // How long to wait for Preload to work before falling back to iframe method
+        LOADER_TIMEOUT = 3000;
+
+    // Tells the browser to execute the Preloaded script by adding it to the DOM
+    function promote() {
+        var script = document.createElement("script");
+        script.id = "boomr-scr-as";
+        script.src = boomerangUrl;
+
+        // Not really needed since dynamic scripts are async by default and the script is already in cache at this point,
```

Comment:
should we make all of these php comments instead so they don't take up bytes on the page?

Reply:
Many WP installs use a minifier (e.g. W3 Total Cache), which should minify everything and strip comments.

I'd like to keep the snippet consistent with what we have everywhere else for easier comparison and making changes.